DQM/EcalMonitorTasks/src/PNDiodeTask.cc

```diff
@@ -59,7 +60,15 @@ namespace ecaldqm {
     }
 
     std::for_each(_ids.begin(), _ids.end(), [&](EcalElectronicsIdCollection::value_type const& id){
-        set->fill(EcalElectronicsId(id.dccId(), id.towerId(), 1, id.xtalId()));
+        try {
+          set->fill(EcalElectronicsId(id.dccId(), id.towerId(), 1, id.xtalId()));
+        }
+        catch(cms::Exception e) {
```

Comment:
hi @tanmaymudholkar - please avoid the try- catch block here. An if statement should be sufficient.

Reply:
Hi @davidlange6 , I've thought about this and I'm not quite sure if there's an easy solution.

Briefly, the DQM crash can be traced through the DQM code until this line: https://github.com/cms-sw/cmssw/blob/master/DQM/EcalCommon/src/MESetBinningUtils2.cc#L504

which fails here: https://github.com/cms-sw/cmssw/blob/master/Geometry/EcalMapping/src/EcalElectronicsMapping.cc#L475

To implement your suggestion and avoid a try-catch block, I can copy and paste this code snippet: https://github.com/cms-sw/cmssw/blob/master/Geometry/EcalMapping/src/EcalElectronicsMapping.cc#L427-L473

... and then put an if condition on ieta. However, this won't be very readable.

The other option would be to figure out which values of dcc, tower, strip, and channel IDs lead to the errors we have been seeing, but that may turn out to be a little complicated -- for instance, I don't know if the allowed range for one of these IDs depends on the other values.

I should mention that the reason for this fix (which has been deployed already in production) is a very rare problem (i.e. events with this issue are seen once every few hundred lumisections). For my curiosity, why should we avoid try-catch blocks? Is it code consistency, readability, or efficiency? The latter two should not be particularly affected by this fix...

Reply:
Its a rare problem, but we suffer the consequence of try-catch on each event. Its code performance question...

anyway I realize I don't understand the idea of this code - you have a EcalElectronicsId and its used to make another EcalElectronicsId, changing its stripId. Then the xtalId() of that new EcalElectronicsId is added to the histogram. That transform sees to be kind of a no-op. How about

set->fill(id.xtalId())

?